extensions/wikia/UserProfilePageV3/tests/UserTagsStrategyTest.php

```diff
@@ -0,0 +1,196 @@
+	 * @param bool $isBlocked Whether this user is blocked
+	 * @param bool $isChatBanned Whether this user is banned from chat
+	 * @param bool $isFounder Whether this user is the founder
+	 * @param array $expectedTags Array of strings (expected tags and their content)
+	 * @covers UserTagsStrategy::getUserTags
+	 * @dataProvider getUserTagsDataProvider
+	 */
+	public function testGetUserTags( array $globalGroups, array $localGroups, $isBlocked, $isChatBanned, $isFounder, array $expectedTags ) {
+		$permissionsServiceMock = $this->getMock( PermissionsServiceImpl::class, [ 'getExplicitGlobalGroups' , 'getExplicitLocalGroups' ] );
+		$permissionsServiceMock->expects( $this->any() )
+			->method( 'getExplicitGlobalGroups' )
+			->willReturn( $globalGroups );
+		$permissionsServiceMock->expects( $this->any() )
+			->method( 'getExplicitLocalGroups' )
+			->willReturn( $localGroups );
+		$container = ContainerBuilder::buildDevContainer();
+		$container->set( PermissionsService::class, $permissionsServiceMock );
+		$injector = new Injector( $container );
+		Injector::setInjector( $injector );
+
+		$chatUserMock = $this->getMockBuilder( ChatUser::class )
+			->disableOriginalConstructor()
+			->setMethods( [ 'isBanned' ] )
+			->getMock();
+		$chatUserMock->expects( $this->any() )
+			->method( 'isBanned' )
+			->willReturn( $isChatBanned );
+		$this->mockClass( ChatUser::class, $chatUserMock );
+
+		$userMock = $this->getMock( User::class, [ 'isBlocked', 'getId' ] );
+		$userMock->expects( $this->any() )
+			->method( 'isBlocked' )
+			->willReturn( $isBlocked );
+		$userMock->expects( $this->any() )
+			->method( 'getId' )
+			->willReturn( 1 );
+
+		$this->mockStaticMethod( WikiFactory::class, 'getWikiById', (object)[ 'city_founding_user' => (int) $isFounder ] );
+
+		$strategy = new UserTagsStrategy( $userMock );
+		$this->assertEquals( $expectedTags, $strategy->getUserTags() );
+	}
+
+	public function getUserTagsDataProvider() {
+		return [
+			'normal user' => [ [], [], false, false, false,
+				[]
+			],
+			'blocked user' => [ [], [], true, false, false,
+				[ wfMessage( 'user-identity-box-group-blocked' )->escaped() ]
+			],
+			'blocked user with other rights' => [ [ 'helper' ], [ 'sysop' ], true, true, true,
+				[ wfMessage( 'user-identity-box-group-blocked' )->escaped() ]
+			],
+			'blocked staff member' => [ [ 'staff' ], [], true, false, false,
+				[ wfMessage( 'user-identity-box-group-staff' )->escaped() ]
+			],
+			'blocked staff member with other rights' => [ [ 'staff' ], [ 'sysop' ], true, false, false,
+				[
+					wfMessage( 'user-identity-box-group-staff' )->escaped(),
+					wfMessage( 'user-identity-box-group-sysop' )->escaped()
+				]
+			],
+			'staff founder' => [ [ 'staff' ], [ 'bureaucrat', 'sysop' ], false, false, true,
+				[
+					wfMessage( 'user-identity-box-group-staff' )->escaped(),
+					wfMessage( 'user-identity-box-group-founder' )->escaped()
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "user-identity-box-group-founder" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)